Banners GA (#9656)
braze-inc/braze-docs: _docs/_user_guide/message_building_by_channel/banners/create.md

---
nav_title: Creating a Banner
article_title: Creating a Banner
page_order: 1
description: "This reference article covers how to create, compose, configure and send Banners using Braze campaigns."
tool:
  - Campaigns
channel:
  - banners
---

# Creating a Banner campaign

> This page covers how to create Banners when you build a campaign in Braze.

## Prerequisites

These are the minimum SDK versions to start using Banners:

{% sdk_min_versions swift:11.3.0 android:33.1.0 web:5.8.1 reactnative:14.0.0 flutter:13.0.0 %}

## Creating a Banner campaign

### Step 1: Create placements

If you haven’t already, you’ll need to create Banner placements in Braze that are used to define the locations in your app or site that can display Banners. To create a placement, go to **Settings** > **Banners Placements**, then select **Create placement**.

> Give your placement a name and assign a Placement ID. Be sure you consult other teams before assigning an ID, as it’ll be used throughout the card’s lifecycle and shouldn’t be changed later. For more information, see Placement IDs.

Reviewer: *fixed locally

When a user clicks a link in the Banner, you can choose to navigate them deeper into your app or redirect them to another webpage. Additionally, you can choose to [log a custom attribute or event]({{site.baseurl}}/developer_guide/analytics/), which will update your user's profile with custom data when they click the Banner.

{% alert important %}
{::nomarkdown}
On-click behavior can be overridden if a specific element (such as a button, link, or image, of the banner) has its own on-click behavior. For example, given the following on-click behaviors:<br><br><ul><li>A Banner has an on-click behavior that redirects to a website's homepage.</li><li>An image in the Banner has an on-click behavior that redirects to a website's product page.</li></ul>If a user clicks the image, they'll be redirected to the product page. However, clicking the surrounding area in the Banner will redirect them to the homepage.
{:/}
{% endalert %}

### Step 4: Set campaign duration

Choose a start date and time for your Banner campaign. By default, Banners last indefinitely. You can change this by selecting End Time and specifying an end date and time.

### Step 5: Set Banner priority

> When multiple campaigns reference the same placement ID, Banners are displayed in order of priority level. By default, newly created Banners are set to medium. If multiple Banners share the same priority level, the newest Banner will be displayed first.

Reviewer 2: ```suggestion
When multiple campaigns reference the same placement ID, Banners are displayed in order of priority level. Newly created Banners are assigned a medium priority by default and positioned above existing medium priority Banners.
```

There is never a situation where Banners share the same exact priority. They can be in the same priority bucket, and by default, new Banners (i.e., the newest Banner) are assigned the top slot in the medium bucket. 

What is displayed is the highest priority Banner for a placement (+ given eligibility). Just want to make that distinction.

Reviewer: @mark-at-braze 

would a better way to phrase it be "priority type"? want to avoid the word bucket (as i'm guessing you feel similar given you didn't add it in your suggestion lol).

or is the easiest fix just to keep it as is but add that banners can share priority levels? like:

```
When multiple campaigns reference the same placement ID, Banners are displayed in order of priority level: high, medium, or low. By default, newly created Banners are set to medium.

Banners can be set to the same priority level. When this is the case, the newest Banner for that priority level is displayed first.
```

the reason why i didn't add your suggestion as-is is because i'm assuming that this logic applies to all priority levels? i.e. if there's 3 high, 2 medium, 4 low, then each will be displayed in order from high to low > newest to oldest?

You can manually set the priority to high, medium, or low.

> To set the priority for a Banner:

Reviewer 2: ```suggestion
To set the exact priority for a Banner:
```

1. Select **Set exact priority**.
2. Drag and drop the campaigns to order them with the correct priority.
3. Select **Apply Sort**.

If you have multiple Banner campaigns using the same placement ID, we recommend using the drag-and-drop priority sorter to define the exact priority.

{% alert note %}
As you create more Banner campaigns, a new Banner's priority is set as the highest priority in the **Medium Priority** section of the exact priority sorter.

> {% endalert %}

Reviewer 2: I worked this bit into the opening paragraph of this section. I don't have a strong opinion if is should exist there vs this alert. Can you remove/alter one of them? Dealers choice.

### Step 6: Finish building the campaign

Finish building your campaign by completing the following:

| Option | Description |
| --- | --- |
| **Targeted Users** | Target users by choosing segments or filters to narrow down your audience. You’ll automatically be given a snapshot of the approximate segment population. Exact segment membership is calculated just before the message is sent. |
| **Conversion Events** | Track how often users perform specific actions after receiving a campaign. You can define conversion events with up to a 30-day window to count the action as a conversion. |
{: .reset-td-br-1 .reset-td-br-2 role="presentation" }

### Step 7: Test and launch

After building your campaign, test and review it to make sure your campaign works as expected. When you’re ready, launch your Banner campaign!

> ## Frequently asked questions

Reviewer 2: To add...
When are Banners personalized?
Banner personalization is refreshed automatically at the start of each new session. There's no need to resend or update the banner campaign—personalized content will update based on the latest customer data every time a new session begins.

When is campaign priority determined for a user within a placement?
If a user qualifies for multiple banner campaigns in the same placement, the banner from the highest-priority campaign will be chosen at the session start.

Reviewer: *added locally

Reviewer: @mark-at-braze quick follow up: for first question

are you meaning: "When does a Banner update after making changes?"

### What are the dimensions of a Banner?

> The HTML will take up the full width of the container it’s rendered in. To learn more about Banner specifications, see About Banners.

Reviewer 2: Link somewhere?

### Can I create multiple Banners using the same placement?

> Only one Banner can be used for each placement ID. However, up to 10 Banners can be assigned to a placement.

Reviewer 2: The Q and the A aren't for each other.

Q. Can I assign the same Banner to multiple placements?
A. A Banner can only be assigned to one placement ID. However, up to 10 Banners can be assigned to a placement.

> ### Can Banners be dismissed?

Reviewer 2: ```suggestion
### Can Banners be dismissed by the user?
```

> No. When a user is no longer eligible for a segment targeted by a Banner campaign, they won't see it again on their next session.

Reviewer 2: ```suggestion
No, users cannot manually dismiss Banners. However, you can control Banner visibility by managing user segment eligibility. When a user no longer meets the targeting criteria for a Banner campaign, they won't see it again on their next session.

For example, if you display a promotional Banner until a user makes a purchase, logging an event such as `purchase_completed` can remove that user from the targeted segment, effectively hiding the Banner in subsequent sessions.
```

### Can I export Banners campaign analytics using the Braze API?

> Yes. You can use the [`/campaigns/data_series` endpoint]({{site.baseurl}}/api/endpoints/export/campaigns/get_campaign_analytics/) to get data on how many Banner campaigns were viewed, clicked, or converted.

Reviewer 2: Does this FAQ belong in this section? Maybe in reporting?

### When are users segmented?

> Users are segmented at the beginning of the session. If a campaign's targeted segments depend on custom attributes, custom events, or other targeting attributes, they must be present on the user at the at the beginning of the session. 

Reviewer 2: ```suggestion
Users are segmented at the beginning of the session. If a campaign's targeted segments depend on custom attributes, custom events, or other targeting attributes, they must be present on the user at the beginning of the session. 
```

### How can I compose Banners to ensure the lowest latency?

> The simpler the messaging in your Banner, the faster it will render. It’s best to test your Banner campaign against the expected latency for your use case. For example, be sure to test Liquid attributes like `catalog_items`.

Reviewer 2: We need to be more specific here. Latency increases because of (1) the amount of content (e.g., more images, scripts) and (2) the complexity of the logic. 

@niccrowley any nuance we should include here?

Reviewer 3: We need the nuance but It's a hard thing to communicate. We can enhance post GA once we get more data on usage patterns

Reviewer: @mark-at-braze @niccrowley , fyi i meant to respond earlier but been heads down: +1 on keeping it vague for now and updating over the next week or so. i'll be and stand by for banner support, so just ping me in channel as needed 🫡

### What Liquid tags are not supported?

The following Liquid tags aren’t supported: Connected Content, promotion codes, and catalog `:rerender` tags.

### How can click events be captured?

> Click events are only captured if an on-click action is set on a `logClick` element and is called using the JS Bridge.

Reviewer 3: This one was asked by a CSM so probs worth including:

If I update a banner, when will those updates appear for users?

Banners can be updated at any time. User will receive the updated banner the next time they start a session.

Reviewer: @niccrowley , is this more or less the same question here? 

https://github.com/braze-inc/braze-docs/pull/9656#discussion_r2096182089

Reviewer 3: > How can click events be captured?

This is about clicks which is not the same as https://github.com/braze-inc/braze-docs/pull/9656#discussion_r2096182089 which talks about when users are segmented vs this one which is talking about what's needed to capture click events

Reviewer: @niccrowley sorry, once again should have been more clear. to clarify, the comment i linked has this at the top:

```
### When are Banners personalized?

Banner personalization is refreshed automatically at the start of each new session. There's no need to resend or update the banner campaign—personalized content will update based on the latest customer data every time a new session begins.

```

a few messages up you wrote:

```
### If I update a banner, when will those updates appear for users?

Banners can be updated at any time. User will receive the updated banner the next time they start a session.
```

are these saying the same things in different word? currently they feel very similar to me but could truly just be my lack of total understanding.

Reviewer 3: ahhh sorry Yes they're the same thing

#### Step 3.1: Style the Banner

You can drag and drop blocks and rows into the canvas area to start building your message.

To customize your message's background properties, border settings, and more, select **Styles**. If you only want to customize the style for a specific block or row, select it to make changes.

![Style panel of the Banner composer.]({% image_buster /assets/img/banners/banner_card_styles.png %})

#### Step 3.2: Define on-click behavior

When a user clicks a link in the Banner, you can choose to navigate them deeper into your app or redirect them to another webpage. Additionally, you can choose to [log a custom attribute or event]({{site.baseurl}}/developer_guide/analytics/), which will update your user's profile with custom data when they click the Banner.

Note that the on-click behavior can be overridden if a specific element, such as a button, link, or image, of the banner has its own on-click behavior. For example, you could have the following on-click behaviors:

- A Banner has an on-click behavior that redirects to a website's homepage.
- An image in the Banner has an on-click behavior that redirects to a website's product page.

> This means if a user clicks the image, they'll be redirected to the product page, but clicking the surrounding area in the Banner will redirect them to the homepage.

Reviewer 3: IMO their should be a note that click events are only triggered on elements that have click actions. If not click actions are defined no clicks will be tracked for the campaign

Reviewer 4: Yeah, calling this out in bold will avoid PQs

Finish building your campaign by completing the following:

| Option | Description |
| --- | --- |
| **Targeted Users** | Target users by choosing segments or filters to narrow down your audience. You’ll automatically be given a snapshot of the approximate segment population. Exact segment membership is calculated just before the message is sent. |
| **Conversion Events** | Track how often users perform specific actions after receiving a campaign. You can define conversion events with up to a 30-day window to count the action as a conversion. |
{: .reset-td-br-1 .reset-td-br-2 role="presentation" }

### Step 7: Test and launch

After building your campaign, test and review it to make sure your campaign works as expected. When you’re ready, launch your Banner campaign!

## Frequently asked questions

### What are the dimensions of a Banner?

The HTML will take up the full width of the container it’s rendered in. To learn more about Banner specifications, see About Banners.

### Can I create multiple Banners using the same placement?

Only one Banner can be used for each placement ID. However, up to 10 Banners can be assigned to a placement.

### Can Banners be dismissed?

> No. When a user is no longer eligible for a Banner, they won't see it again on their next session.

Reviewer 3: ```suggestion
No. When a user is no longer segmented for a Banner campaign, they won't see it again on their next session.
```

> No. When a user is no longer eligible for a Banner, they won't see it again on their next session.

Reviewer 3: ```suggestion
No. When a user is no longer eligible for a segment targeted by a Banner campaign, they won't see it again on their next session.
```

Reviewer 4: I think I prefer the original version. If you include a reference to segments, you should also include a reference to filters, as both apply. 

Being removed from the segment is not the only time a banner will stop appearing. For example, it will stop appearing in a given placement if any of the following take place:
*`abort_message()` is called during /sync
* A higher priority banner is launched
* the priority of the existing banner campaign is lowered, causing a different banner campaign to show

### Can I export Banners campaign analytics using the Braze API?

> Yes. You can use the [`/campaigns/data_series` endpoint]({{site.baseurl}}/api/endpoints/export/campaigns/get_campaign_analytics/) to get data on how many Banner campaigns were sent, opened, clicked, or converted.

Reviewer 3: @m-larue @chrishaines @cgiulietti-braze @exigoking 

Has anyone tested this? I'm not certain this is supported.

Reviewer 4: Yes. See this [slack thread](https://brazetechnology.slack.com/archives/C07E1P1LT0W/p1744827787785919?thread_ts=1744812709.206379&cid=C07E1P1LT0W)

> Yes. You can use the [`/campaigns/data_series` endpoint]({{site.baseurl}}/api/endpoints/export/campaigns/get_campaign_analytics/) to get data on how many Banner campaigns were sent, opened, clicked, or converted.

Reviewer 3: ```suggestion
Yes. You can use the [`/campaigns/data_series` endpoint]({{site.baseurl}}/api/endpoints/export/campaigns/get_campaign_analytics/) to get data on how many Banner campaigns were viewed, clicked, or converted.
```

We do not track sent

### When are users segmented?

> Users are segmented at the beginning of the session. If a campaign's targeted segments depend on custom attributes, custom events, or other targeting attributes need to be ingested into Braze and present on the user at the time of segmentation. 

Reviewer 3: ```suggestion
Users are segmented at the beginning of the session. If a campaign's targeted segments depend on custom attributes, custom events, or other targeting attributes,  these attributes must be present on the user at the at the beginning of the session. 
```

### How can I compose Banners to ensure the lowest latency?

The simpler the messaging in your Banner, the faster it will render. It’s best to test your Banner campaign against the expected latency for your use case. For example, be sure to test Liquid attributes like `catalog_items`.

### What Liquid tags are not supported?

> The following Liquid tags aren’t supported: Connected Content, promotion codes, and catalog `:rerender` tags.

Reviewer 5: It is the `catalog_items` tag specifically. I think it could either be referred to as "catalog_items with the `:rerender` option" or "catalog items with liquid re-rendering"

Reviewer 6: +1

Reviewer: qq @chrishaines : is `rerender` option like a braze-specific thing? i was only able to find `render` on the [liquid docs site](https://shopify.dev/docs/api/liquid/tags/render).

cc @niccrowley

Reviewer: (i wanted to link to the tag in their official docs so readers could learn more if they wanted)

Reviewer 3: Yea it specific to catalog Items since we allow liquid in catalog items rerender needs to be set in order to render the liquid

Existing docs: https://www.braze.com/docs/user_guide/personalization_and_dynamic_content/catalogs/using_catalogs/#templating-catalog-items:~:text=Templating%20catalog%20items%20including%20Liquid

Reviewer: AYE! thanks 🙌🏽

### How can click events be captured?

> Click events are only captured if an on-click action is set on a `logClick` element and is called using the JS Bridge.

Reviewer 3: ```suggestion
Click events are only captured if an on-click action is set on the editor block or `logClick` is called in a custom html block via the JS Bridge.
```

Reviewer 4: What's the "editor block"?

Reviewer: @niccrowley , just following up here when you get the chance 👍🏽

Reviewer 3: https://www.braze.com/docs/user_guide/message_building_by_channel/in-app_messages/drag_and_drop/editor_blocks

These are editor blocks, you could probably just link out to this tho may be confusing cause it's within the IAM docs. I'll file a docs request to move these editor blocks doc to a generic space.

Reviewer: ahh, thanks for the deets. i'll figure out how best to refer to this.
